packages/typespec-azure-core/src/rules/no-union-expr.ts

```diff
@@ -0,0 +1,63 @@
+import { createRule, paramMessage, Program, Union } from "@typespec/compiler";
+import { $ } from "@typespec/compiler/typekit";
+import { getHeaderFieldName, isHeader, isStatusCode } from "@typespec/http";
+
+export const noUnionExprRule = createRule({
+  name: "no-union-expr",
+  description: "Azure services should not define union expression but create a declaration.",
+  severity: "warning",
+  url: "https://azure.github.io/typespec-azure/docs/libraries/azure-core/rules/no-union-expr",
+  messages: {
+    default: paramMessage`Union expression should be defined as a named union declaration.`,
+  },
+  create(context) {
+    const program = context.program;
+    const excludedUnions = new Set<Union>();
+    const invalidUnions = new Set<Union>();
+
+    return {
+      modelProperty: (prop) => {
+        const type = prop.type;
+        if (type.kind !== "Union" || type.name) {
+          return;
+        }
+
+        if (
+          isStatusCode(program, prop) ||
+          (isHeader(program, type) &&
+            getHeaderFieldName(program, type).toLowerCase() === "content-type")
```

Comment:
looks like contentType is not coming back with 'content-type' as header name for some reason.